keps/sig-node/1287-in-place-update-pod-resources/README.md

```diff
@@ -454,162 +486,167 @@ added. This ensures that resizes don't affect previously admitted existing Pods.
 
-T=9: Kubelet watching the pod sees resize #3 and accepts it
-    - kubelet sends patch {
-        `resourceVersion` = `<previous value>` # enable conflict detection
-        `status.containerStatuses[0].allocatedResources[cpu]` = 1.6
-        `status.resize[cpu]` = "InProgress"'
-      }
+1. Kubelet syncs the pod, and sees resize #3 and admits it
     - `spec.containers[0].resources.requests[cpu]` = 1.6
     - `status.containerStatuses[0].allocatedResources[cpu]` = 1.6
-    - `status.resize[cpu]` = "InProgress"
+    - `acknowledged[cpu]` = 1.5
     - `status.containerStatuses[0].resources.requests[cpu]` = 1.5
+    - `status.conditions[type==PodResizePending]` removed
+    - `status.conditions[type==PodResizing]` added
+    - actual CPU shares = 1536
 
-T=10: Container runtime applied cpu=1.6
-    - kubelet sends patch {
-        `resourceVersion` = `<previous value>` # enable conflict detection
-        `status.containerStatuses[0].resources.requests[cpu]` = 1.6
-        `status.resize[cpu]` = unset
-      }
+1. Container runtime applied cpu=1.6
     - `spec.containers[0].resources.requests[cpu]` = 1.6
     - `status.containerStatuses[0].allocatedResources[cpu]` = 1.6
-    - `status.resize[cpu]` = unset
+    - `acknowledged[cpu]` = 1.6
+    - `status.containerStatuses[0].resources.requests[cpu]` = 1.5
+    - `status.conditions[type==PodResizing]`
+    - actual CPU shares = 1638
+
+1. Kubelet syncs the pod
+    - `spec.containers[0].resources.requests[cpu]` = 1.6
+    - `status.containerStatuses[0].allocatedResources[cpu]` = 1.6
+    - `acknowledged[cpu]` = 1.6
     - `status.containerStatuses[0].resources.requests[cpu]` = 1.6
+    - `status.conditions[type==PodResizing]` removed
+    - actual CPU shares = 1638
 
-T=11: Resize #4: cpu = 100
+1. Resize #4: cpu = 100
     - apiserver validates the request and accepts the operation
-    - apiserver sets `resize[cpu]` to "Proposed"
     - `spec.containers[0].resources.requests[cpu]` = 100
     - `status.containerStatuses[0].allocatedResources[cpu]` = 1.6
-    - `status.resize[cpu]` = "Proposed"
+    - `acknowledged[cpu]` = 1.6
     - `status.containerStatuses[0].resources.requests[cpu]` = 1.6
+    - actual CPU shares = 1638
 
-T=12: Kubelet watching the pod sees resize #4
-    - this node does not have 100 CPUs, so kubelet cannot accept
-    - kubelet sends patch {
-        `resourceVersion` = `<previous value>` # enable conflict detection
-        `status.resize[cpu]` = "Infeasible"'
-      }
+1. Kubelet syncs the pod, and sees resize #4
+    - this node does not have 100 CPUs, so kubelet cannot admit it
     - `spec.containers[0].resources.requests[cpu]` = 100
     - `status.containerStatuses[0].allocatedResources[cpu]` = 1.6
-    - `status.resize[cpu]` = "Infeasible"
+    - `acknowledged[cpu]` = 1.6
     - `status.containerStatuses[0].resources.requests[cpu]` = 1.6
-```
+    - `status.conditions[type==PodResizePending].type` = `"Infeasible"`
+    - actual CPU shares = 1638
```

Comment:
This timeline is great.  So great I think you should do more (non-blocking).

I'd love to see the situation where an NRI plugin mutates the request to round to 10 shares, just to demonstrat how actual and allocated are related (e.g. I assume the actual would show shares/1024, so that 1.5 request would translate to 1536 shares, which gets rounded to 1540, which comes to actual as 1.504.  At that point, the scheduler, with its `max()` logic would consider 1.504, not 1.500.  Right?

Reply:
Will address in a separate PR.